src/datta_lab_to_nwb/markowitz_gillis_nature_2023/metadatainterface.py

```diff
@@ -0,0 +1,37 @@
+"""Primary class for handling metadata non-specific to any other DataInterfaces."""
+from neuroconv.basedatainterface import BaseDataInterface
+from neuroconv.utils import load_dict_from_file
+from pynwb import NWBFile
+
+
+class MetadataInterface(BaseDataInterface):
```

Comment:
I see what you're trying to do here, but I think it can have greater effect if you utilize OOP by either

(i) having those interfaces that remove this `get_metadata` protocol inherit from this as a parent

(ii) defining this as a mix-in (requires a little bit of know-how to do properly)

(iii) or define this protocol only once inside the main NWB Converter class, which is where top level metadata like this is generally extracted

Reply:
As we discussed today, opting for a conversion-specific BaseDataInterface so that each DataInterface can operate stand-alone